src/WooCommerce.php

```diff
@@ -128,6 +128,32 @@ public static function woocommerce_single_product_image_thumbnail_html($html, $t
     return preg_replace('/(<a\s+)/i', '<a data-srcset="' . $srcset . '" data-sizes="' . $srcsizes . '" ', $html);
   }
 
+  /**
+   * Adds additional images from product variation gallery.
+   *
+   * @implements woocommerce_product_get_gallery_image_ids
+   */
+  public static function woocommerce_product_get_gallery_image_ids($value, $obj) {
+    global $product;
+
+    if ($product->get_type() === 'variable') {
+      $variations = $product->get_available_variations();
+      $images = [];
+      // Add images from variation gallery.
+      foreach ($variations as $variation) {
+        $images[] = $variation['image_id'];
+        $variation_gallery = get_post_meta(
+          $variation['variation_id'],
+          '_' . Plugin::PREFIX . '_variation_gallery',
```

Comment:
As the meta field appears on a variation product, it doesn't make sense to have "variation" in the meta field name. And as the plugin name is already "gallerya", it doesn't make sense to repeat "gallery" in the meta field name. The name should rather orient itself at the core meta field "image_id".
```suggestion
          '_gallerya_image_ids',
```
```suggestion
          '_' . Plugin::PREFIX . '_image_ids',
```

Or if we support more than only images:

```suggestion
          '_' . Plugin::PREFIX . '_attachment_ids',
```
```suggestion
          '_' . Plugin::PREFIX . '_media_ids',
```